design/proposals/proxy-support/proxy-support.md

```diff
@@ -0,0 +1,157 @@
+# Maintaining Requester IP Addresses <!-- omit in toc --> 
+
+When Conjur (or DAP) are run behind a load balancer, users are unable to use CIDR restrictions on Hosts or Layers. Furthermore, Load Balancer IP address appears in Audit logs instead of the true client IP address. This is because Conjur references the last entry in the `X-Forwarded-For` header.
+
+This decision was conscious, and meant to prevent IP spoofing. We have reached a point where customers need a mechanism to add known proxies to Conjur/DAP to identify the true identity of the requesting client. 
+
+This design document is meant to provide a possible solution to the challenge of identifying the client IP while preventing IP spoofing.
+
+- [Current Request Flow](#current-request-flow)
+- [Proposed Solution (Summary)](#proposed-solution-summary)
+- [Proposed Solution (Detailed)](#proposed-solution-detailed)
+  - [Adding Proxies with Policy](#adding-proxies-with-policy)
+  - [Limitations](#limitations)
+    - [Proxies cannot be layered](#proxies-cannot-be-layered)
+- [Rejected Alternatives](#rejected-alternatives)
+  - [Configuration using Environment Variables](#configuration-using-environment-variables)
+
+## Current Request Flow
+
+The following visualization offers a typical experience as a requests moves through a Layer 7 load balancers.
+
+![](/design/diagrams/out/proxy-support-overview/proxy-support-overview.png)
+
+As we can see, each load balancer adds its IP address to the `X-Forwarded-For` header.
+
+With our current implementation:
```

Comment:
Where is code located ? (i cant find it)

Reply:
https://github.com/cyberark/conjur/blob/0a724b7f3c0af0ae681a0261ad87d4aaa6b1fab0/app/controllers/authenticate_controller.rb#L55